sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/VectorIndexSpec.java

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.models;
+
+import com.azure.cosmos.implementation.Constants;
+import com.azure.cosmos.implementation.JsonSerializable;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+/**
+ * Vector Indexes spec for Azure CosmosDB service.
+ */
+public final class VectorIndexSpec {
+
+    private final JsonSerializable jsonSerializable;
+    private String type;
+
+    /**
+     * Constructor
+     *
+     * @param path the path.
+     */
+    public VectorIndexSpec(String path) {
+        this.jsonSerializable = new JsonSerializable();
+        this.setPath(path);
+    }
+
+    /**
+     * Constructor.
+     */
+    public VectorIndexSpec() {
+        this.jsonSerializable = new JsonSerializable();
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param objectNode the object node that represents the included path.
+     */
+    public VectorIndexSpec(ObjectNode objectNode) { this.jsonSerializable = new JsonSerializable(objectNode); }
+
+    /**
+     * Gets path.
+     *
+     * @return the path.
+     */
+    public String getPath() {
+        return this.jsonSerializable.getString(Constants.Properties.PATH);
```

Comment:
Is it not inefficient to use the `jsonSerializable` to access every property each time the getter method is called? Should we have class properties to hold the values until we need to serialize the model?

Reply:
JsonSerializable has a Map to access the properties - so, it is more expensive than returning a variable but avoids the sync of the variable with the JsonSerializable dictionary - and since these model classes are just a structured wrapper around a Json object, it think this is fine - we have never seen perf issues there in other cases. And this class in only ever used when a customer creates new containers - so, no need to change this IMO.